mne/io/edf/edf.py

```diff
@@ -134,36 +133,33 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
         read_size = len(r_lims) * buf_len
         with open(self._filenames[fi], 'rb', buffering=0) as fid:
             # extract data
-            fid.seek(data_offset +
-                     block_start_idx * buf_len * n_chan * data_size)
+            start_offset = (data_offset +
+                            block_start_idx * buf_len * n_chan * data_size)
+            ch_offsets = np.cumsum(np.concatenate([[0], n_samps])) * data_size
```

Comment:
I know this was my line, but to save one multiply the `* data_size` can go onto `n_samps * data_size` instead of outside the concat and cumsum